BeeTracker/app/src/main/java/com/blueprint/foe/beetracker/Model/StorageAccessor.java

```diff
@@ -38,15 +49,104 @@ private String convertStreamToString(InputStream is) throws IOException {
         return sb.toString();
     }
 
-    public Submission load(InputStream inputStream) throws IOException {
+    public static Submission load(InputStream inputStream) throws IOException {
         String string = convertStreamToString(inputStream);
         inputStream.close();
         Gson gson = new Gson();
         return gson.fromJson(string, Submission.class);
     }
 
-    public Submission load(Context context) throws IOException {
+    public static Submission load(Context context) throws IOException {
         InputStream fis = context.openFileInput(FILENAME);
         return load(fis);
     }
+
+    public static String saveBitmapExternally(Bitmap bitmap) throws FileNotFoundException {
+        File pictureFile = getOutputMediaFile(MEDIA_TYPE_IMAGE);
+        FileOutputStream out = null;
+
+        try {
+            out = new FileOutputStream(pictureFile);
+            bitmap.compress(Bitmap.CompressFormat.JPEG, 100, out);
+            return pictureFile.getAbsolutePath();
+        } finally {
+            try {
+                if (out != null) {
+                    out.close();
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    public static String saveBitmapInternally(Bitmap bitmap, Context context) throws FileNotFoundException {
+        FileOutputStream fos = null;
+        try {
+            fos = context.openFileOutput(TEMPORARY_IMAGE_FILENAME, Context.MODE_PRIVATE);
```

Comment:
Is it possible to have multiple temporary images?

Reply:
The temporary image is only being used between picture capture and permission to save externally being granted. If a user takes another picture without granting permission, it will override their old picture.